yaptide/celery/tasks.py

```diff
@@ -0,0 +1,181 @@
+                    'unit': str(axis.unit),
+                    'name': str(axis.name),
+                    'values': axis.data.tolist(),
+                }
+
+            elif page.dimension == 2:
+                page_dict['data']['values'] = page.data_raw.tolist()
+
+                axis: MeshAxis = page.plot_axis(0)
+                page_dict['first_axis'] = {
+                    'unit': str(axis.unit),
+                    'name': str(axis.name),
+                    'values': axis.data.tolist(),
+                }
+                axis: MeshAxis = page.plot_axis(1)
+                page_dict['second_axis'] = {
+                    'unit': str(axis.unit),
+                    'name': str(axis.name),
+                    'values': axis.data.tolist(),
+                }
+
+            else:
+                # Add info about the location of the file containging to many dimensions
+                raise ValueError(f'Invalid number of pages {page.dimensions}')
+
+            est_dict['pages'].append(page_dict)
+        result_dict['estimators'].append(est_dict)
+
+    return result_dict
+
+
+@celery_app.task
+def simulation_task_status(task_id: str) -> dict:
+    """Task responsible for returning simulation status"""
+    task = AsyncResult(id=task_id, app=celery_app)
+
+    result = {
+        'status': 'OK',
+        'message': {
+            'state': task.state
+        }
+    }
+    if task.state == "PENDING":
+        result['message']['status'] = 'Pending...'
+    elif task.state == "PROGRESS":
+        result['message']['status'] = 'Calculations in progress...'
+        sim_info = sim_status_from_logfile(path_to_file=task.info.get('path')+'/run_1/shieldhit0001.log')
+        result['message']['info'] = sim_info
+    elif task.state != 'FAILURE':
+        result['message']['status'] = task.info.get('status', '')
+        if 'result' in task.info:
+            result['message']['result'] = task.info.get('result')
+    else:
+        result['status'] = 'ERROR'
+        result['message']['status'] = str(task.info)
+
+    return result
+
+
+def sim_status_from_logfile(path_to_file: str):
+    """Extracts current simulation state from simulation logfile"""
+    # This is dummy version because pymchelper currently doesn't privide any information about progress
+    with open(path_to_file, 'r') as reader:
+        found_line_which_starts_status_block = False
+        last_result_line = ""
+        for line in reader:
+            if not found_line_which_starts_status_block:
+                # We are searching for lines containing progress info
+                # They are preceded by line starting with "Starting transport"
+                found_line_which_starts_status_block = line.lstrip().startswith("Starting transport")
+            else:
+                # Searching for latest line
+                if line.lstrip().startswith("Primary particle"):
+                    last_result_line = line
+
+        splited = last_result_line.split()
```

Comment:
What if we call this method on a file which was generated in the early phase of simulation. It will not contain any of ""Starting transport" and "Primary particle" lines.. Then `last_result_line` will be empty string and you will not be able to call `splitted[3]`

Reply:
I'll add some if statement